mergify_engine/clients/github.py

```diff
@@ -43,6 +43,11 @@ class TooManyPages(Exception):
     response: httpx.Response
 
 
+@dataclasses.dataclass
+class GraphqlError(Exception):
+    response: httpx.Response
```

Comment:
The typing looks wrong to me, this is not a `httpx.Response`, but a dictionary. 
Since the error format is not documented by GitHub, just put `str` for now.